x-pack/elastic-agent/_meta/config/common.p2.yml.tmpl

```diff
@@ -10,10 +10,14 @@ outputs:
 
 inputs:
   - type: system/metrics
+
+    # The only two requirement are that it has only characters allowed in an Elasticsearch index name
+    # and does NOT contain a `-`.
```

Comment:
I think it would be better to be explicit on what is allowed even if it is conservative in nature. That way its less likely for them to make a mistake and its clear directly from the comment.

How about:

`Must be all lowercase, no spaces, only characters a-z or numbers 0-9 (no special characters), and no more than 255 characters.`

I think the character length might be an issue being its concat-ed to make the index name.